bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/InboxAutoIgnore.java

```diff
@@ -0,0 +1,142 @@
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private ThingRegistry thingRegistry;
+    private Inbox inbox;
+    private boolean autoIgnore = true;
+
+    public InboxAutoIgnore() {
+        super(ThingStatusInfoChangedEvent.TYPE);
+    }
+
+    @Override
+    public void receiveTypedEvent(ThingStatusInfoChangedEvent event) {
+        if (autoIgnore) {
+            if (ThingStatus.ONLINE.equals(event.getStatusInfo().getStatus())) {
+                checkAndIgnoreInInbox(event);
+            } else if (ThingStatus.REMOVING.equals(event.getStatusInfo().getStatus())) {
+                removePossiblyIgnoredResultInInbox(event);
+            }
+        }
+    }
+
+    private void checkAndIgnoreInInbox(ThingStatusInfoChangedEvent event) {
+        Thing thing = thingRegistry.get(event.getThingUID());
+        if (thing != null) {
+            ThingHandler handler = thing.getHandler();
+            if (handler != null) {
+                String deviceUid = handler.getDeviceId();
+                if (deviceUid != null) {
+                    ignoreInInbox(deviceUid);
+                }
+            }
+        }
+    }
+
+    private void ignoreInInbox(String deviceUid) {
+        List<DiscoveryResult> results = inbox.get(InboxFilterCriteria.representationFilter(deviceUid, null));
+        if (results.size() == 1) {
+            logger.debug("Auto-ignoring the inbox entry for the device uid {}", deviceUid);
+            inbox.setFlag(results.get(0).getThingUID(), DiscoveryResultFlag.IGNORED);
+        }
+    }
+
+    private void removePossiblyIgnoredResultInInbox(ThingStatusInfoChangedEvent event) {
+        Thing thing = thingRegistry.get(event.getThingUID());
+        if (thing != null) {
+            ThingHandler handler = thing.getHandler();
+            if (handler != null) {
+                String deviceUid = handler.getDeviceId();
+                if (deviceUid != null) {
+                    removeFromInbox(deviceUid);
+                }
+            }
+        }
+    }
+
+    private void removeFromInbox(String deviceUid) {
+        List<DiscoveryResult> results = inbox
+                .get(InboxFilterCriteria.representationFilter(deviceUid, DiscoveryResultFlag.IGNORED));
+        if (results.size() == 1) {
+            logger.debug("Removing the ignored result from the inbox for the device uid {}", deviceUid);
+            inbox.remove(results.get(0).getThingUID());
+        }
+    }
+
+    @Activate
+    protected void activate(Map<String, String> properties) {
+        if (properties != null) {
+            String value = properties.get("autoIgnore");
+            autoIgnore = value == null || !value.toString().equals("false");
+        }
+    }
+
+    @Reference
+    protected void setThingRegistry(ThingRegistry thingRegistry) {
+        this.thingRegistry = thingRegistry;
+    }
+
+    protected void unsetThingRegistry(ThingRegistry thingRegistry) {
```

Comment:
Yes, it is. No need to annotate this method as well, but looks strange, I agree.